File: 백준/Bronze/34323. 할인이 필요해/할인이 필요해.c
#include <stdio.h>

int main(){
    long long n, m, s;
    scanf("%lld %lld %lld", &n, &m, &s);

    long long n_discount = s * (m + 1) * (100 - n) / 100.0;
    long long m_discount = s * m;
    
    printf("%lld", n_discount < m_discount ? n_discount : m_discount);

    return 0;
}

File: 백준/Bronze/34323. 할인이 필요해/할인이 필요해.py
import sys

def need_discount():
    n, m, s = map(int, sys.stdin.readline().split())

    n_discount = s * (m + 1) * (100 - n) // 100 #(s - (s * n/100)) * (m+1)
    m_discount = s * m

    return min(n_discount, m_discount)

if __name__ == "__main__":
    print(int(need_discount()))